### PR TITLE
fix comments/indent

### LINKED ISSUE #7: Do not change the indentation
I ran this tool against some of my code bases using the command provided in the README:

`find . -name "*.go" | xargs -I{} go-instrument -app my-service -w -filename {}`

It seems to change the indentation of various files, even though they are already properly formatted using `go fmt`. A git diff of one of those files:

```diff
index 6da35ba..8c5f832 100644
--- a/defaults.go
+++ b/defaults.go
@@ -10,58 +10,58 @@ import (
 )

 const (
-    logFormatJSON = "json"
+    logFormatJSON   = "json"

-    logTargetStdout = "stdout"
-    logTargetStderr = "stderr"
+    logTargetStdout = "stdout"
+    logTargetStderr = "stderr"

-    logLevelPanic = "panic"
-    logLevelFatal = "fatal"
-    logLevelError = "error"
-    logLevelWarn  = "warn"
-    logLevelInfo  = "info"
-    logLevelDebug = "debug"
-    logLevelTrace = "trace"
+    logLevelPanic   = "panic"
+    logLevelFatal   = "fatal"
+    logLevelError   = "error"
+    logLevelWarn    = "warn"
+    logLevelInfo    = "info"
+    logLevelDebug   = "debug"
+    logLevelTrace   = "trace"
 )

// ...
```

### PR DESCRIPTION
**Pull request by burik666**
It's a dirty hack, but it works.

Fixes #7 #8

### CODE REVIEW

**Comment by nikolaydubina** [comment]
nice. if this is what I think it is, it is amazing and definitely should be merged. let me read into it a bit. (probably will a little adjust OOP later on)

File: processor/processor.go [hunk 1]
```diff
@@ -157,15 +159,20 @@ func (p *Processor) Process(fset *token.FileSet, file ast.File) error {
 			return true
 		}
 
-		fn.Body.List = append(p.Instrumenter.PrefixStatements(spanName, hasError), fn.Body.List...)
-
-		c.Replace(fn)
+		ps := p.Instrumenter.PrefixStatements(spanName, hasError)
+		patches = append(patches, patch{pos: int(fn.Body.Pos()), stmts: ps})
```
Comment (nikolaydubina) [inline processor/processor.go:163]: better to pass dedicated type. it is a bit confusing what this is actually.
```suggestion
		patches = append(patches, patch{pos: fn.Body.Pos(), stmts: ps})
```

File: processor/patch.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import (
+	"bytes"
+	"go/ast"
+	"go/format"
+	"go/parser"
+	"go/token"
+)
+
+type patch struct {
+	pos   int
```
Comment (nikolaydubina) [inline processor/patch.go:12]:
```suggestion
	pos   token.Pos
```

File: processor/patch.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		buf.WriteString("\n")
+
+		pos := patch.pos + offset
+		src = append(src[:pos], append(buf.Bytes(), src[pos:]...)...)
+		offset += buf.Len()
+	}
+
+	p.updateFile(fset, file, src)
+
+	return nil
```
Comment (nikolaydubina) [inline processor/patch.go:41]: ignored error.
```suggestion
	return p.updateFile(fset, file, src)
```

File: processor/patch.go [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+func formatNode(fset *token.FileSet, node any) ([]byte, error) {
+	var buf bytes.Buffer
+	if err := format.Node(&buf, fset, node); err != nil {
+		return nil, err
+	}
+
+	return buf.Bytes(), nil
+}
+
+func (p *Processor) updateFile(fset *token.FileSet, file *ast.File, newSrc []byte) error {
```
Comment (nikolaydubina) [inline processor/patch.go:53]: no need for method receiver, right? + renaming what is happening
```suggestion
func replaceFileWithSrc(fset *token.FileSet, file *ast.File, newSrc []byte) error {
```

File: processor/patch.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+		src = append(src[:pos], append(buf.Bytes(), src[pos:]...)...)
+		offset += buf.Len()
+	}
+
+	p.updateFile(fset, file, src)
+
+	return nil
+}
+
+func formatNode(fset *token.FileSet, node any) ([]byte, error) {
```
Comment (nikolaydubina) [inline processor/patch.go:44]:
```suggestion
func formatNodeToBytes(fset *token.FileSet, node any) ([]byte, error) {
```

File: processor/patch.go [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+
+func (p *Processor) patchFile(fset *token.FileSet, file *ast.File, patches ...patch) error {
+	src, err := formatNode(fset, file)
+	if err != nil {
+		return err
+	}
+
+	var buf bytes.Buffer
+
+	offset := -int(file.FileStart) + 1
```
Comment (nikolaydubina) [inline processor/patch.go:24]: why it is negative though? better for clarity keep it positive and substract when you need relative.
```suggestion
	offsetOriginalFile := int(file.FileStart) + 1
```

File: processor/patch.go [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+	for _, patch := range patches {
+		buf.Reset()
+
+		buf.WriteString("\n")
+		if err := format.Node(&buf, fset, patch.stmts); err != nil {
+			return err
+		}
+		buf.WriteString("\n")
+
+		pos := patch.pos + offset
```
Comment (nikolaydubina) [inline processor/patch.go:34]: so this is position in file?
```suggestion
		posInFile := patch.pos - offset
```

File: processor/patch.go [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+		src = append(src[:pos], append(buf.Bytes(), src[pos:]...)...)
```
Comment (nikolaydubina) [inline processor/patch.go:35]:
```suggestion
		src = append(src[:posInFile], append(buf.Bytes(), src[posInFile:]...)...)
```

File: processor/patch.go [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+		offset += buf.Len()
```
Comment (nikolaydubina) [inline processor/patch.go:36]:
```suggestion
                // reducing offset position in original file by size of current modification
		offset -= buf.Len()
```

**Review by nikolaydubina — COMMENTED** [review]
I think it is good. need to apply comments for clarity, it is hard to grasp what is happening. can keep same algorithm and structure. feel free to update, or else I will update myself this in coming week or so

File: processor/patch.go (hunk 4 above)
Comment (burik666) [reply processor/patch.go:53]: Also `processor.functionName`, `processor.isContext`, etc.. methods do not need a receiver.

I think we’ll leave it as it is, in the future we’ll probably store fset inside the processor.

File: main.go [hunk 10]
```diff
@@ -21,37 +22,53 @@ func main() {
...
 		defaultSelect bool
 		skipGenerated bool
+		gofmt         bool
 	)
 	flag.StringVar(&fileName, "filename", "", "go file to instrument")
 	flag.StringVar(&app, "app", "app", "name of application")
 	flag.BoolVar(&overwrite, "w", false, "overwrite original file")
 	flag.BoolVar(&defaultSelect, "all", true, "instrument all by default")
 	flag.BoolVar(&skipGenerated, "skip-generated", false, "skip generated files")
+	flag.BoolVar(&gofmt, "gofmt", false, "format file (gofmt)")
```
[the remark this answers is not in the document]
Comment (burik666) [reply main.go:32]: I thought we shouldn't do anything other than instrumentation.

File: processor/patch.go (hunk 6 above)
Comment (burik666) [reply processor/patch.go:24]: offset changes during the process, here it is the starting point.
In most cases it is always 1

Comment (nikolaydubina) [reply processor/patch.go:24]: yeah, really need to make this positive. negative file offset is just triggering security alarms bells..

File: processor/patch.go [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+		pos := int(patch.pos) + offset
+		src = append(src[:pos], append(buf.Bytes(), src[pos:]...)...)
+		offset += buf.Len()
+	}
+
+	return p.updateFile(fset, file, src)
+}
+
+func formatNodeToBytes(fset *token.FileSet, node any) ([]byte, error) {
+	var buf bytes.Buffer
```
Comment (nikolaydubina) [inline processor/patch.go:43]: probably some smarter reuse of buffers possible between files.. but let's merge even as is for now.

**Pull request merged**